### DISCUSSION TITLE
What is domain ? How is it different from workspace ?

### DISCUSSION THREAD

**Discussion by aniketgm** [wezterm/wezterm#4324] [Q&A] [answered]
Wezterm has lots of features in it. I trying to understand one such feature, `Domain`. I couldn't find a clear explanation in the docs.

As the title says, what exactly is this domain ? In the docs I see we have SSHDomain, WSLDomain, etc, but how is it different as opposed to creating a workspace of SSH and/or WSL, etc.

Can anyone please explain ?

**Comment by wez** [comment, marked as answer]
we could do with some docs that talk about this explicitly.
Here's a brief summary:

A domain is really an *execution domain*, a way to reason about how to run commands in a certain place.
By default, wezterm starts up and runs programs in the `local` execution domain. Programs are started on the local machine, and are spawned directly by the wezterm process using the native PTY system.

The various other domains allow you to run programs via a multiplexer server (using the SSH, TLS and unix domains), in WSL containers using the WSL domain, or in some other customizable context (such as docker or systemd) using [ExecDomains](https://wezfurlong.org/wezterm/config/lua/ExecDomain.html).

The domain allows wezterm to know how to spawn programs in that environment, and allows eg: starting new panes with the same current working directory from the active pane if the new pane belongs to the same domain.

You don't need to use domains in wezterm if you don't want to; you can manually run commands via ssh, docker, wsl and so on and they will work just fine.  However, wezterm won't be able to intelligently reconcile things like the current working directory when it doesn't know about the execution domain.

Workspaces are essentially a window management feature.  A window is tagged with the active workspace label when it is created, unless you explicitly set its workspace differently when it is spawned.

When you change the active workspace in the GUI, the windows from the current workspace are hidden and the windows from the newly activated workspace are shown instead.

**Reply by aniketgm** [reply]
Thank you for this details explanation. It cleared my confusion.
